src/code/z_play.c

```diff
@@ -53,7 +56,47 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167DE4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167F0C.s")
+void func_80167F0C(PlayState* this) {
+    // Draw pause menu
+    if ((this->pauseCtx.state != 0) || (this->pauseCtx.debugEditor != 0)) {
+        KaleidoScopeCall_Draw(this);
+    }
+
+    // Draw interface/HUD
+    if (gSaveContext.gameMode == 0) {
+        func_8011F0E0(this);
+    }
+
+    // Draw Message items
+    if (((this->pauseCtx.state == 0) && (this->pauseCtx.debugEditor == 0)) || (this->msgCtx.currentTextId != 0xFF)) {
+        func_80156758(this);
+    }
```

Comment:
```suggestion
void Play_DrawOverlayElements(PlayState* this) {
    if ((this->pauseCtx.state != 0) || (this->pauseCtx.debugEditor != DEBUG_EDITOR_NONE)) {
        KaleidoScopeCall_Draw(this);
    }

    if (gSaveContext.gameMode == 0) {
        Interface_Draw(this);
    }

    if (((this->pauseCtx.state == 0) && (this->pauseCtx.debugEditor == DEBUG_EDITOR_NONE)) || (this->msgCtx.currentTextId != 0xFF)) {
        Message_Draw(this);
    }
```
If you want to include this function here, might as well name what we already know

Reply:
I wouldn't say the shrinking part is part of the overlay elements, though, I think we should have a more general name.

Reply:
I'm actually pretty close to PRing play, even if just as a WIP. So if it isn't too much hassle I'd prefer to deal with this function in that PR.

Reply:
https://github.com/zeldaret/mm/pull/1109#discussion_r989401508

Reply:
Right, I'll undo it.